### PR TITLE
Make the 'add assessment' button a dropdown like the designs

### PR DESCRIPTION
**Pull request by slav-petkov-saorsa**
Background:
https://trello.com/c/Y51nfQC5/439-make-the-add-assessment-button-a-dropdown-like-the-designs

Solution:
image

### CODE REVIEW

**Comment by slav-petkov-saorsa** [comment]
[Make the 'add assessment' button a dropdown like the designs](https://trello.com/c/Y51nfQC5/439-make-the-add-assessment-button-a-dropdown-like-the-designs)

File: app/javascript/packs/assessments.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-document.querySelectorAll('.need--status_complete').forEach(e => e.style.display = 'none');
+Array.prototype.slice.call(document.querySelectorAll('.need--status_complete')).forEach((e, i) => e => e.style.display = 'none');
```
Comment (timwis) [inline app/javascript/packs/assessments.js:1]: You don't need to wrap this in a slice; [`NodeList` has a `forEach` method](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach).

File: features/step_definitions/assessment_steps.rb [hunk 2]
```diff
@@ -4,12 +4,14 @@
 
 When('I choose to schedule an assessment') do
   @assessment_type = 'schedule'
-  click_link 'Schedule an assessment'
+  page.find('#assessmentDropdownElements').click
```
Comment (jamiebuckley) [inline features/step_definitions/assessment_steps.rb:7]: This breaks if you run in a JS browser, e.g. firefox
Have pushed up https://github.com/wearefuturegov/beacon/tree/fix/assessment_tests_browser
We can leave it until after this release as it's not critical and the tests are passing in the pipeline

File: app/views/contacts/show.html.erb [hunk 3]
```diff
@@ -69,13 +69,21 @@
...
+            <div id="assessmentDropdownElements" class="assessment-dropdown-content">
+              <%= link_to('Schedule an assessment', new_contact_assessment_path(@contact.id, :type => 'schedule'), id: 'schedule-assesment-btn') %>
+              <%= link_to('Log an assessment', new_contact_assessment_path(@contact.id, :type => 'log'), id: 'log-assesment-btn') %>
+            </div>
+          </div>
+          <noscript>
+            <ul style="list-style-type: none" id="schedule_an_assessment">
+            <li><%= link_to('Schedule an assessment', new_contact_assessment_path(@contact.id, :type => 'schedule'), id: 'schedule-assesment-nojs-btn') %></li>
+            <li><%= link_to('Log an assessment', new_contact_assessment_path(@contact.id, :type => 'log'), id: 'log-assesment-nojs-btn') %></li>
+          </ul>-->
```
Comment (timwis) [inline app/views/contacts/show.html.erb:83]: I think you may have accidentally left a `-->` on this line.

File: app/javascript/packs/assessments.js (hunk 1 above)
Comment (slav-petkov-saorsa) [reply app/javascript/packs/assessments.js:1]: But doesn't work in IE(there NodesList doesn't support forEach)

File: features/step_definitions/assessment_steps.rb (hunk 2 above)
Comment (timwis) [inline features/step_definitions/assessment_steps.rb:7]: Shouldn't this be clicking `#btnAssessmentDropdown` to expand the dropdown? And wouldn't that only work on a `@javascript` enabled test anyway?

File: app/views/contacts/show.html.erb (hunk 3 above)
Comment (slav-petkov-saorsa) [reply app/views/contacts/show.html.erb:83]: Will fix it, yhanks

File: app/javascript/packs/assessments.js (hunk 1 above)
Comment (timwis) [reply app/javascript/packs/assessments.js:1]: ohhhh! ok

File: features/step_definitions/assessment_steps.rb (hunk 2 above)
Comment (rik86future) [reply features/step_definitions/assessment_steps.rb]: sorted

File: app/views/contacts/show.html.erb (hunk 3 above)
Comment (rik86future) [reply app/views/contacts/show.html.erb:83]: sorted

**Pull request merged**